src/azure-cli/azure/cli/command_modules/acs/_params.py

```diff
@@ -440,6 +462,39 @@ def load_arguments(self, _):
         c.argument('acr', validator=validate_registry_name)
         c.argument('node_name')
 
+    with self.argument_context('aks maintenanceconfiguration') as c:
+        c.argument('cluster_name', help='The cluster name.')
+
+    for scope in ['aks maintenanceconfiguration add', 'aks maintenanceconfiguration update']:
+        with self.argument_context(scope) as c:
+            c.argument('config_name', options_list=[
+                       '--name', '-n'], help='The config name.')
+            c.argument('config_file', help='The config json file.')
+            c.argument('weekday', help='Weekday on which maintenance can happen. e.g. Monday')
```

Comment:
Do we need to consider defining this parameter as an enumeration type?

Reply:
It can be made an enumeration however in the azure-cli-extension, it is being used as shown above. Do you think changing it is appropriate?

Reply:
Okay, it can align with the logic of the extension first